src/test/java/com/sap/cloudfoundry/client/facade/CloudControllerClientIntegrationTest.java

```diff
@@ -105,20 +137,410 @@ void deleteDomainMissing() throws IOException {
+                                                                             .getName());
+        } finally {
+            client.deleteApplication(applicationName);
+        }
+    }
+
+    private CloudPackage uploadApplication(String applicationName) throws URISyntaxException, InterruptedException {
+        URL url = getClass().getResource("staticfile.zip");
+        Path path = Paths.get(url.toURI());
+        CloudPackage cloudPackage = client.asyncUploadApplication(applicationName, path);
+        while (cloudPackage.getStatus() != Status.READY && !hasFailed(cloudPackage.getStatus())) {
+            cloudPackage = client.asyncUploadApplication(applicationName, path);
+            Thread.sleep(1000);
+        }
+        if (hasFailed(cloudPackage.getStatus())) {
+            fail(MessageFormat.format("Cloud package is in invalid state: \"{0}\"", cloudPackage.getStatus()));
+        }
+        return cloudPackage;
+    }
+
+    private boolean hasFailed(Status status) {
+        return status == Status.EXPIRED || status == Status.FAILED;
+    }
+
+    private void stageApplication(String applicationName, CloudPackage cloudPackage) throws InterruptedException {
+        UUID applicationGuid = client.getApplicationGuid(applicationName);
+        CloudBuild build = createBuildForPackage(cloudPackage);
+        client.bindDropletToApp(build.getDropletInfo()
+                                     .getGuid(),
+                                applicationGuid);
+    }
+
+    private CloudBuild createBuildForPackage(CloudPackage cloudPackage) throws InterruptedException {
+        CloudBuild build = client.createBuild(cloudPackage.getGuid());
+        while (build.getState() == CloudBuild.State.STAGING) {
+            build = client.getBuild(build.getGuid());
+            Thread.sleep(1000);
+        }
+        if (build.getState() == CloudBuild.State.FAILED) {
+            fail(MessageFormat.format("Cloud build is in invalid state: \"{0}\"", build.getState()));
+        }
+        return build;
+    }
+
+    private void verifyApplicationWillBeCreated(String applicationName, Staging staging, Set<CloudRouteSummary> cloudRoutesSummary) {
+        try {
+            client.createApplication(applicationName, staging, DISK_IN_MB, MEMORY_IN_MB, cloudRoutesSummary);
+            assertApplicationExists(ImmutableCloudApplication.builder()
+                                                             .name(applicationName)
+                                                             .diskQuota(DISK_IN_MB)
+                                                             .memory(MEMORY_IN_MB)
+                                                             .instances(1)
+                                                             .routes(cloudRoutesSummary)
+                                                             .build());
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
```

Comment:
Do we need to catch that exception? The test will fail anyway.

Reply:
You're right, done